config/v1/types_infrastructure.go

```diff
@@ -1519,7 +1519,15 @@ type IBMCloudServiceEndpoint struct {
 
 // IBMCloudPlatformSpec holds the desired state of the IBMCloud infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type IBMCloudPlatformSpec struct{}
+type IBMCloudPlatformSpec struct {
+	// serviceEndpoints is a list of custom endpoints which will override the default
+	// service endpoints of an IBM Cloud service. These endpoints are consumed by
+	// components within the cluster to reach the respective IBM Cloud Services.
```

Comment:
I think we should add some description here of what happens when you add values. They get verified and then copied to status by some controller right? And then consumed by?